reframe/frontend/loader.py

```diff
@@ -133,7 +133,7 @@ def load_from_module(self, module):
             getlogger().debug('No tests registered')
             return []
 
-        candidates = module._rfm_gettests()
+        candidates = module._rfm_gettests(*args, **kwargs)
```

Comment:
And this is the line that requires some updates on the test registry. The only reason why the argument forwarding goes on beyond this point, is because this function returns the instances of the class, and with its current implementation there is no way to retrieve the classes. So I think that instead of forwarding these arguments, we should just make the test registry a class that the loader understands and can access the regression test classes in the registry before explicitly calling an `instantiate_all` function. Hence, if we can access the classes, we can just set the variable defaults here with a simple `setattr`.